### PR TITLE
functionalities in junos-ansible being merged from roles to collection.

### PR DESCRIPTION
**Pull request by rahkumar651991**
Changes done to call PyEz functions to handle reboot, shutdown and similar functionalities in junos-ansible being merged from roles to collection.

### CODE REVIEW

File: ansible_collections/Juniper/junos_collection/plugins/modules/juniper_junos_system.py [hunk 1]
```diff
@@ -285,163 +285,72 @@ def main():
...
-    # other_re option only applies to reboot, shutdown, or halt actions.
-    if (params.get('other_re') is True and
-       action != 'reboot' and
-       action != 'shutdown' and
-       action != 'halt'):
-        junos_module.fail_json(msg='The other_re option can only be used when '
-                                   'the action option has the value "reboot", '
-                                   '"shutdown", or "halt".')
+    #Four actions are expected - reboot, shutdown, halt and zeroize
+    if action not in {'reboot', 'shutdown', 'halt'}:
```
Comment (dineshbaburam91) [inline ansible_collections/Juniper/junos_collection/plugins/modules/juniper_junos_system.py:299]: Use List here ['reboot', 'shutdown', 'halt']

Comment (rahkumar651991) [reply ansible_collections/Juniper/junos_collection/plugins/modules/juniper_junos_system.py]: modified in latest commit.

File: ansible_collections/Juniper/junos_collection/plugins/modules/juniper_junos_system.py [hunk 2]
```diff
@@ -285,163 +285,72 @@ def main():
...
+        # at, in_min and other_re option only applies to reboot, shutdown, or halt action.
+        for str,val in {"at":at,"in_min":in_min,"other_re":other_re}:
```
[the remark this answers is not in the document]
Comment (rahkumar651991) [reply ansible_collections/Juniper/junos_collection/plugins/modules/juniper_junos_system.py]: modified in latest commit

File: ansible_collections/Juniper/junos_collection/plugins/modules/juniper_junos_system.py [hunk 3]
```diff
@@ -285,163 +285,72 @@ def main():
...
-                else:
-                    # This is the else clause of the for loop.
-                    # It only gets executed if the loop finished without
-                    # hitting the break.
-                    results['msg'] = 'Did not find expected RPC response.'
-                    results['changed'] = False
+            junos_module.logger.debug("Executing RPC")
+
+            if action == 'reboot':
+                got = junos_module.sw.reboot(in_min, at, all_re, None, False, other_re)
```
[the remark this answers is not in the document]
Comment (rahkumar651991) [reply ansible_collections/Juniper/junos_collection/plugins/modules/juniper_junos_system.py]: added vmhost support for juniper_junos_system.

File: ansible_collections/Juniper/junos_collection/plugins/modules/juniper_junos_software.py [hunk 4]
```diff
@@ -708,67 +708,23 @@ def main():
...
                 else:
-                    results['msg'] += ' Reboot successfully initiated.'
+                    # This is the else clause of the for loop.
+                    # It only gets executed if the loop finished without
+                    # hitting the break.
+                    results['msg'] += ' Did not find expected response ' \
+                                      'from reboot RPC. RPC response is ' \
+                                      '%s' % \
+                                      junos_module.etree.tostring(resp)
+                    junos_module.fail_json(**results)
```
Comment (vnitinv) [inline ansible_collections/Juniper/junos_collection/plugins/modules/juniper_junos_software.py:727]: add junos_module.pyez_exception.RpcTimeoutError to line 741

**Pull request merged**